utils/VastParser.js

```diff
@@ -198,6 +227,309 @@
 
       return result;
     }, this);
+    /**
+     * This should be the first thing that happens in the parser function: check if the vast XML has no ads.
+     * If it does not have ads, track error urls
+     * @public
+     * @method VastParser#checkNoAds
+     * @param {XMLDocument} vastXML Contains the vast ad data to be parsed
+     * @param {object} ads A jQuery object which contains the collection of ad elements found
+     * @returns {boolean} true if there are no ads, false otherwise.
+     */
+    this.checkNoAds = function(vastXML, ads) {
+      // if there are no ads in ad response then track error
+      if (ads.length === 0) {
+        OO.log("VAST: No ads in XML");
+        // there could be an <Error> element in the vast response
+        var noAdsErrorURL = $(vastXML).find("Error").text();
+        if (noAdsErrorURL) {
+          this.pingURL(this.ERROR_CODES.WRAPPER_NO_ADS, noAdsErrorURL);
+        }
+        // if the ad response came from a wrapper, then go up the chain and ping those error urls
+        //this.trackError(this.ERROR_CODES.WRAPPER_NO_ADS, this.wrapperParentId);
+        return true;
+      }
+      return false;
+    };
+
+    /**
+     * Helper function to ping error URL. Replaces error macro if it exists.
+     * @public
+     * @method VastParser#pingURL
+     * @param {number} code Error code
+     * @param {string} url URL to ping
+     */
+    this.pingURL = function(code, url) {
+      url = url.replace(/\[ERRORCODE\]/, code);
+      OO.pixelPing(url);
+    };
+
+    /**
+     * Helper function to ping error URLs.
+     * @public
+     * @method VastParser#pingURLs
+     * @param {number} code Error code
+     * @param {string[]} urls URLs to ping
+     */
+    this.pingURLs = function(code, urls) {
```

Comment:
Does this only ping error URLs? We can update the function name to pingErrorURLs to make this easier to understand.